src/main/java/seedu/address/ui/body/address/PersonListPanel.java

```diff
@@ -31,17 +34,30 @@ public class PersonListPanel extends UiPart<Region> {
 
     @FXML
     private ListView<PersonListCellData> personListView;
-
     private int selectedIndex;
 
+    private List<PersonListCardData> allData;
+
+    private List<PersonListCardData> favData;
+
+    private final Logic logic;
+
+    private final ResultDisplay resultDisplay;
+
     /**
      * Creates a {@code PersonListPanel} with the given {@code ObservableList}.
      */
-    public PersonListPanel(ObservableList<Person> personList, PersonDetailPanel panel) {
+    public PersonListPanel(ObservableList<Person> personList, PersonDetailPanel panel, Logic logic,
+                           ResultDisplay resultDisplay) {
         super(FXML);
+        this.allData = null;
+        this.favData = null;
+        this.logic = logic;
+        this.resultDisplay = resultDisplay;
         personListView.setCellFactory(listView -> new PersonListCell());
         personListView.setFocusTraversable(false);
         personListView.setOnMouseClicked(event -> {
```

Comment:
Also, this listener will have to inform the model of changes in selection by the user clicking, so that the info propagates to all the relevant UI components. Something like this within the callback:

```java
Person selectedPerson = personListView.getSelectionModel().getSelectedItem().getPerson();
logic.setSelectedPerson(selectedPerson);
```